tests/serializers/test_natural.py

```diff
@@ -250,6 +256,78 @@ def fk_as_pk_natural_key_not_called(self, format):
         self.assertEqual(obj.object.pk, o1.pk)
 
 
+def natural_key_opt_out_test(self, format):
+    # Refs #35729
+    """
+    Tests natural key opt-out scenarios for models inheriting
+    from AbstractBaseUser.
+
+    Verifies that when natural_key() returns None or (self.pk,):
+    """
```

Comment:
This is a fun one: our coding style docs advises against "tests..." "verifies that..." [preambles](https://docs.djangoproject.com/en/5.2/internals/contributing/writing-code/coding-style/#:~:text=preambles%20such%20as%20%E2%80%9CTests%20that%E2%80%9D%20or%20%E2%80%9CEnsures%20that%E2%80%9D.).